cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -462,6 +475,35 @@ func (o *ScaleUpOrchestrator) ComputeExpansionOption(
 		return option
 	}
 
+	autoscalingOptions, err := nodeGroup.GetOptions(o.autoscalingContext.NodeGroupDefaults)
+	if err != nil {
+		klog.Errorf("Failed to get autoscaling options for node group %s: %v", nodeGroup.Id(), err)
+	}
+	if autoscalingOptions != nil && autoscalingOptions.AtomicScaleUp {
```

Comment:
How is this whole logic supposed to work with similar node groups? As I can see it you are currently taking only the main node group into account, totally ignoring the similar node groups. Current scale-up logic will distribute these new nodes between all the similar node groups, so I don't think you'll get the effect you're looking for.

Reply:
You are right. I have changed the similar node group calculation for atomic node groups to not find similar node groups.